src/Tools/CsvImporter/Program.cs

```diff
@@ -1,251 +1,116 @@
-}
-catch (Exception)
+if (!Directory.Exists(configDirPath))
 {
-    throw new ArgumentException("Missing CSV file path argument");
+    Directory.CreateDirectory(configDirPath);
 }
 
-string maxTasksArg;
-try
+if (!Directory.Exists(logsDirPath))
 {
-    maxTasksArg = args[1];
-}
-catch (Exception)
-{
-    maxTasksArg = "256";
+    Directory.CreateDirectory(logsDirPath);
 }
 
-int maxTasks;
-try
-{
-    maxTasks = int.Parse(maxTasksArg);
-}
-catch (Exception ex)
-{
-    throw new ArgumentException($"Invalid max tasks argument: {maxTasksArg}", ex);
-}
+string configFilePath = Path.Combine(configDirPath, "config.json");
+string logFilePath = Path.Combine(logsDirPath, $"CsvImporter_{DateTime.UtcNow:yyyy-MM-dd_HH-mm-ss}.log");
 
-// Read config file.
-CsvImporterConfig csvImporterConfig;
-try
-{
-    csvImporterConfig = await ConfigFileUtils.GetCsvImporterConfigAsync();
-}
-catch (Exception ex)
-{
-    throw new Exception($"Error reading config file: {ex.Message}");
-}
+bool configFileExists = File.Exists(configFilePath);
 
-QueueClientService queueClientService;
-try
+CsvImporterConfig? preLaunchConfig = null;
+if (configFileExists)
 {
-    string storageConnectionString = Environment.GetEnvironmentVariable("STORAGE_CONNECTION_STRING") ?? throw new NullReferenceException("STORAGE_CONNECTION_STRING environment variable not set");
-
-    queueClientService = new(
-        connectionString: storageConnectionString
+    preLaunchConfig = JsonSerializer.Deserialize(
+        json: File.ReadAllText(configFilePath),
+        jsonTypeInfo: CoreJsonContext.Default.CsvImporterConfig
     );
 }
-catch (NullReferenceException)
-{
-    ConsoleUtils.WriteWarning("STORAGE_CONNECTION_STRING environment variable not set.");
-    ConsoleUtils.WriteInfo("Attempting to use an Azure token credential instead.");
 
-    string queueUriString = csvImporterConfig.QueueUri ?? Environment.GetEnvironmentVariable("QUEUE_URI") ?? throw new NullReferenceException("QUEUE_URI environment variable not set or missing from config file.");
+var builder = Host.CreateApplicationBuilder(args);
+
+builder.Configuration
+    .AddEnvironmentVariables();
+
+builder.Logging
+    .ClearProviders()
+    .AddSimpleConsole(options =>
+    {
+        options.SingleLine = false;
+        options.UseUtcTimestamp = true;
+    })
+    .AddFileLogger(options =>
+    {
+        options.FilePath = logFilePath;
+    });
 
-    TokenCredential tokenCredential;
+if (preLaunchConfig is not null)
+{
     try
     {
-        tokenCredential = new ChainedTokenCredential(
-            new AzureCliCredential(),
-            new AzurePowerShellCredential(),
-            new ManagedIdentityCredential()
-        );
-
-        queueClientService = new(
-            queueUri: new(queueUriString),
-            tokenCredential: tokenCredential
-        );
+        builder.Services
+            .AddCsvImporterQueueClientService(
+                connectionString: AuthUtils.GetStorageConnectionString()
```

Comment:
I think I'm going to drop Azure Storage Account connection strings altogether. In the case of how I _"personally"_ see this should work, the system that's running this should have a dedicated identity assigned to it. Even with an on-premises server, registering the server with Azure Arc and utilizing [the managed identity of the server](https://learn.microsoft.com/en-us/azure/azure-arc/servers/managed-identity-authentication) to authenticate to the Azure Storage Account is relatively simple to do.